Tools/rules.bzl

```diff
@@ -49,14 +49,46 @@ def pyx_library(
     # TODO(robertwb): It might be better to only generate the C files,
     # letting cc_library (or similar) handle the rest, but there isn't yet
     # support compiling Python C extensions from bazel.
+    python_cmd = """
+import os
+import sys
+
+n = len(sys.argv)
+for src, dst in zip(sys.argv[1:], sys.argv[1+n//2:]):
+    src_file_name = src.split(".")[0]
+    src = src_file_name + ".so"
+    try:
+        os.rename(src, dst)
+    except Exception as e:
+        print("Failed to rename: " + src + " to " + dst)
+        dir = os.path.dirname(src_file_name)
+        renamed_candidate = False
+        for file_name in os.listdir(dir):
+            file, ext = os.path.splitext(file_name)
+            if ext != ".so":
+                continue
+            if not file.startswith(os.path.basename(src_file_name)):
+                continue
+            file_dot_split = file.split(".")
+            if len(file_dot_split) >= 2 and file_dot_split[-1].startswith("cpython-"):
```

Comment:
I suspect PyPy would argue with this. (i.e. `cpython-` isn't necessarily the test we want)

Reply:
Would it be preferable to have a test that looks like this:

```
Scan the directory
Skip all non .so files
Split by "." token
If ".".join(file[:-2]) == pyx_source_filename then
   Rename so file
```

Reply:
I think that'd be better. Part of the point of the version info string is to disambiguate implementations so restricting it to cpython feels like a mistake.

Unfortunately I don't really know anything about bazel myself so from my point of view it's very hard to usefully review this.

Reply:
Changed to reflect this pseudocode.